core/lib/zksync_core_leftovers/src/lib.rs

```diff
@@ -738,6 +743,33 @@ pub async fn initialize_components(
         .context("add_tee_verifier_input_producer_to_task_futures()")?;
     }
 
+    if components.contains(&Component::DADispatcher)
+        && eth
+            .sender
+            .clone()
+            .context("eth_sender")?
+            .pubdata_sending_mode
+            == PubdataSendingMode::Custom
+    {
```

Comment:
shall we maybe panic if this is misconfigured? In other words, panic if `DADispatcher` is added for non-cusom `PubdataSendingMode`

Reply:
done, added a panic there